modules/meta/arrow-meta-prototype/testing-plugin/src/main/kotlin/arrow/meta/plugin/testing/Compilation.kt

```diff
@@ -0,0 +1,65 @@
+package arrow.meta.plugin.testing
+
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import org.assertj.core.api.Assertions.assertThat
+import java.io.File
+import java.nio.file.Paths
+import io.github.classgraph.ClassGraph
+
+enum class CompilationResult {
+  OK,
+  INTERNAL_ERROR,
+  COMPILATION_ERROR,
+  SCRIPT_EXECUTION_ERROR
+}
+
+data class CompilationData(
+  val sourceFileName: String,
+  val sourceContent: String,
+  val generatedFileContent: String,
+  val generatedClasses: ArrayList<String>,
+  val compilationResult: CompilationResult
+)
+
+fun testCompilation(compilationData: CompilationData) {
+  val kotlinSource = SourceFile.kotlin(compilationData.sourceFileName, compilationData.sourceContent)
+
+  val result = KotlinCompilation().apply {
+    sources = listOf(kotlinSource)
+    classpaths = listOf(
+      // TODO
+      // classpathOf("arrow-annotations:x.x.x")
+      File("../../arrow-annotations/build/libs/arrow-annotations-0.10.1-SNAPSHOT.jar")
+    )
+    pluginClasspaths = listOf(
+      classpathOf("compiler-plugin")
+//      File("../compiler-plugin/build/libs/compiler-plugin.jar")
+    )
+  }.compile()
+
+  assertThat(result.exitCode).isEqualTo(exitCodeFrom(compilationData.compilationResult))
+
+  if (!result.exitCode.equals(KotlinCompilation.ExitCode.OK)) return
+
+  val actualGeneratedFileContent = Paths.get(result.outputDirectory.parent, "sources", "${compilationData.sourceFileName}.meta").toFile().readText()
+  assertThat(actualGeneratedFileContent).isEqualTo(compilationData.generatedFileContent)
+
+  val actualGeneratedClasses = result.generatedFiles.map { it.name }.filter { it.endsWith(".class") }
+  assertThat(actualGeneratedClasses).isEqualTo(compilationData.generatedClasses.map { "$it.class" })
+}
+
+fun contentFromResource(fromClass: Class<Any>, resourceName: String): String = fromClass.getResource("/$resourceName").readText()
```

Comment:
We frequently follow the style:
```
fun declaration(): RetType =
  expressionBody
```
To easily tell apart the function signature from the actual implementation which gets confusing if its all in one line.

Reply:
Absolutely agree. Thanks!